packages/driver/src/cypress/command_queue.ts

```diff
@@ -60,336 +60,322 @@ const commandRunningFailed = (Cypress, state, err) => {
   })
 }
 
-export default {
-  create: (state, timeout, whenStable, cleanup, fail, isCy) => {
-    const queue = $queue.create()
-
-    const { get, slice, at, reset, clear, stop } = queue
-
-    const logs = (filter) => {
-      let logs = _.flatten(_.invokeMap(queue.get(), 'get', 'logs'))
+export class CommandQueue extends Queue<Command> {
+  state: any
```

Comment:
Should this be the GetterSetter type? https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cypress/setter_getter.d.ts

Reply:
The type of `state` function isn't that easy. To make it useful for our development, we need to create a new definition like:

```ts
interface StateFunc {
  (param: 'document') => Document
  (param: 'done') => boolean
 // etc..
}
```

I left it as `any` to do this work in the next PR.